Documentation/operations/upgrade.rst

```diff
@@ -468,6 +468,17 @@ Changed Metrics
 
 * The ``cilium_api_limiter_processed_requests_total`` has now label ``return_code`` to specify the http code of the request.
 
+Hubble CLI
+~~~~~~~~~~
+
+* the ``--cluster`` behavior changed to show flows emitted from nodes outside of
+  the provided cluster name (either coming from or going to the target cluster).
+  This change brings consistency between the ``--cluster`` and ``--namespace``
+  flags and removed the incompatibility between the ``--cluster`` and
+  ``--node-name`` flags. The previous behavior of ``--cluster foo`` can be
+  reproduced with ``--node-name foo/`` (shows all flows emitted from a node in
+  cluster ``foo``).
```

Comment:
Can we make it even more clear that if you use the new Hubble CLI with an older Cilium cluster, that the `--cluster` flag won't work? I don't think that's obvious if you read this and don't know how it's implemented.

Reply:
I'm surprised by the fact that the `Endpoint.Clustername` is only populated with the name of remote clusters, not the local cluster. Is that true?

I would have expected the Hubble parser to also populate the `ClusterName` field to also be set for local clusters, meaning the new version of the flag returns more flows than the old one, not fewer.

Reply:
> I'm surprised by the fact that the Endpoint.Clustername is only populated with the name of remote clusters, not the local cluster. Is that true?

I can indeed repro with a local kind cluster using cilium `latest`: 

```json
    "destination": {
      "ID": 1305,
      "identity": 19193,
      "namespace": "kube-system",
      "labels": [
        "k8s:app.kubernetes.io/name=hubble-relay",
        "k8s:app.kubernetes.io/part-of=cilium",
        "k8s:io.cilium.k8s.namespace.labels.kubernetes.io/metadata.name=kube-system",
        "k8s:io.cilium.k8s.policy.cluster=default",
        "k8s:io.cilium.k8s.policy.serviceaccount=hubble-relay",
        "k8s:io.kubernetes.pod.namespace=kube-system",
        "k8s:k8s-app=hubble-relay"
      ],
      "pod_name": "hubble-relay-6c49d69996-xcq6g",
      "workloads": [
        {
          "name": "hubble-relay",
          "kind": "Deployment"
        }
      ]
    },
```
I checked to be sure that we have the endpoint cluster name in the server's protobuf definition
```
% grpcurl -plaintext localhost:4245  describe .flow.Endpoint
flow.Endpoint is a message:
message Endpoint {
  uint32 ID = 1;
  uint32 identity = 2;
  string namespace = 3;
  repeated string labels = 4;
  string pod_name = 5;
  repeated .flow.Workload workloads = 6;
  string cluster_name = 7;
}
```
So https://github.com/cilium/cilium/pull/32313 is included but not working as expected as we see `k8s:io.cilium.k8s.policy.cluster=default` in the labels but no endpoint's `cluster_name` field cc @rolinh. I took a quick look at the code but didn't find why it's not working.

Reply:
@gandro ok so after further investigation:
> I'm surprised by the fact that the `Endpoint.Clustername` is only populated with the name of remote clusters, not the local cluster. Is that true?

The cluster name should be set regardless of whether the pod is from the local or a remote cluster. With that said, there is currently a bug where cluster name is not set for local endpoints (local to the node), that is why in my last comment attempt the cluster name is not set for Hubble Relay. https://github.com/cilium/cilium/pull/35415 should fix the missing cluster name for local endpoints (amongst other cases).

> I would have expected the Hubble parser to also populate the ClusterName field to also be set for local clusters, meaning the new version of the flag returns more flows than the old one, not fewer.

Conceptually yes, after the patch `--cluster foo` will show flows emitted by nodes outside of cluster `foo` either going to or from cluster `foo`, which `--node-name foo/` doesn't return. Technically, since the patch is based on endpoint labels, the new version will not show reserved identity flows (i.e. health check, node to node etc.). So the new set of returned flows is not a superset of the current version, but filtering out reserved identities flows is also consistent with the namespace filters. Does that make sense?